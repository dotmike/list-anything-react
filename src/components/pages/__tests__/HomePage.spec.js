import React from "react";
import configureStore from "redux-mock-store";

import ConnectedHomePage, { UnconnectedHomePage } from "../HomePage";

describe("ConnectedHomePage", () => {
  const mockStore = configureStore();
  const initialState = {};
  const props = { store: mockStore(initialState) };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(<ConnectedHomePage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("UnconnectedHomePage", () => {
  const props = {
    isAuthenticated: false
  };

  describe("when `isAuthenticated` is false", () => {
    it("renders correctly", () => {
      props.isAuthenticated = false;
      const wrapper = shallowWithContext(<UnconnectedHomePage {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when `isAuthenticated` is true", () => {
    it("renders correctly", () => {
      props.isAuthenticated = true;
      const wrapper = shallowWithContext(<UnconnectedHomePage {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
