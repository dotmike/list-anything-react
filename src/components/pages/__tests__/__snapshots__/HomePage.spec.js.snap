// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedHomePage renders correctly 1`] = `
<HomePage
  dispatch={[Function]}
  isAuthenticated={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`UnconnectedHomePage when \`isAuthenticated\` is false renders correctly 1`] = `
<div>
  <Redirect
    push={false}
    to="/login"
  />
</div>
`;

exports[`UnconnectedHomePage when \`isAuthenticated\` is true renders correctly 1`] = `
<div>
  <Redirect
    push={false}
    to="/dashboard"
  />
</div>
`;
